mavlink/qgroundcontrol
# Utm adapter upstream v1

UTM adapter V1 (https://github.com/Dronecode/utm-adapter)

Description
-----------
In this PR the following things are done

- CMake Build System Updated
- Boost Dependencies Removed 
- QNetworkAccessManager 
- QML Resources Refreshed
- Date-Time Bug Fixed
- Refactor UI 

Test Steps
-----------
To enable and build QGC with UTMSP Adapter support using cmake on CLI 
1. Make sure you cloned the repository and updated the submodules before, see chapter Source Code above and switch into the repository folder: 
```
cd qgroundcontrol
```
2. Configure: 
```
cmake -B build -G Ninja -DCMAKE_BUILD_TYPE=Debug -DQGC_UTM_ADAPTER=ON
```
3. Build: 
```
cmake --build build --config Debug
```
4. Run the QGroundcontrol binary
```
./build/QGroundControl
``` 

Checklist:
----------


Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

https://github.com/user-attachments/assets/2d8efc87-9cdb-4466-b47c-0bc2708b21f8
